### ISSUE TITLE
No requests showing after install

### ISSUE THREAD

**Issue by FozzieHi** [mysociety/alaveteli#7823]
I've just installed the develop branch of alaveteli through Docker using the [installation instructions](https://alaveteli.org/docs/installing/docker/). I then also ran `script/rails-post-deploy`, which as I understand it, runs the database migrations.

I can then log into `Joe Admin` fine, and view all authorities. However, when I go to browse requests, none are displayed. Additionally, when I go to `Requests` in the admin panel I get the following error. Is there anything I'm missing here? There are no errors when I run the tests with `bundle exec rake spec`.
```
undefined method `nil_class_both_links' for #<ActionView::Base:0x00000000028be0>
```
![image](https://github.com/mysociety/alaveteli/assets/23139726/75ba8326-bb89-4cec-b0f2-d80272cd49bb)

**Comment by HelenWDTK** [comment]
AIUI this is caused by the sample data still containing external requests, which are no longer a thing.

Run this to fix it `rails runner 'InfoRequest.external.destroy_all'`

**Comment by gbp** [comment]
Yes as @HelenWDTK said this is due to external requests which we were in the [process to removing from the codebase](https://github.com/mysociety/alaveteli/pull/6899) but priorities have moved on to other features/work.

**Comment by FozzieHi** [comment]
Thanks both, that was indeed the problem for the admin page, and that command fixed it. However, when browsing all requests, or a specific authority page, I still can't see the list of requests.
![image](https://github.com/mysociety/alaveteli/assets/23139726/c09b8b6b-15f3-40a3-8a62-beaca82ba9ab)
![image](https://github.com/mysociety/alaveteli/assets/23139726/cb40fe17-818c-4d01-bcee-9041af669a47)

Here's what I have tried:
- The requests still do not show up when logging into an admin account.
- The requests do show up on the admin pages, both for the Authorities and Requests pages.
- I can view the request if I follow the link directly (both logged in and not logged in).
- If I create a new request, I can't see it on either page.

Given the amount of requests do show up on the authorities page, I'm thinking this might be a caching or anti-spam issue? However, I've taken a look at the config and it seems like caching should be disabled on the development environment, and opening a private window still doesn't show any requests. Is it possible this is some anti-spam issue I'm running into, or am I missing anything else? Thanks!

**Comment by gbp** [comment]
Those pages are powered by the Xapian search index. You'll need to run the `script/destroy-and-rebuild-xapian-index` initially and then `script/update-xapian-index` for any subsequent changes EG after creating new requests or after responses are received.

The requests on admin pages are direct from the database which is why those work.

**Comment by FozzieHi** [comment]
Amazing, that worked. Thanks!

I did run `script/update-xapian-index` from the docs, but must have missed `script/destroy-and-rebuild-xapian-index`. Should I submit a PR to add `rails runner 'InfoRequest.external.destroy_all'` (and I guess also `script/destroy-and-rebuild-xapian-index` in general) to the installation guide in the docs? Or would it be more beneficial to remove external requests from the sample data instead?

**Comment by gbp** [comment]
Not too sure why running `script/destroy-and-rebuild-xapian-index` is necessary as this should happen when [setting up the Docker container](https://github.com/mysociety/alaveteli/blob/232178b63357a15c958db78eeec68b89508cefd7/docker/setup#L42-L44).

We would happily accept PRs for updating the installations guide, this is located on the [Github pages branch](https://github.com/mysociety/alaveteli/tree/gh-pages).

I suspect removing the external requests from the sample fixture data is harder than it seems as unfortunately our tests are very tightly coupled together.

**Comment by FozzieHi** [comment]
I did also have to run the database migrations manually, using the command in the installation guide, as none had been run when I ran the server for the first time.

I'm guessing in my case at least, nothing under `if $data_reset; then` ran for whatever reason? This makes a bit more sense, because I ran the database migrations, loading the sample data, and then `script/update-xapian-index` (missing out `script/destroy-and-rebuild-xapian-index`).
